### DISCUSSION TITLE
Importing with CMake

### DISCUSSION THREAD

**Discussion by hudkinsnoah** [JSBSim-Team/jsbsim#956] [Q&A]
I am trying to import JSBSim files to my source files in my project and cmake is not finding files. It seems like it has to do with my cmake setup and I am not sure what to import in my CMakeLists.txt file. 

JSBSim is a submodule in of the project. I have then setup a "target_link_libraries" and tried adding the JSBSim library or at the very least the src directory since the file not being found is FGFDMExec.h. I have tried both libJSBSim and JSBsim with no luck.

Any help would be much appreciated, thanks!

**Comment by bcoconni** [comment]
Hi @hudkinsnoah,

Sorry for the delay in responding to your request.

You could check the discussion in issue #684 where another user has reported an issue about using JSBSim as a subproject of their own CMake project. In particular, the discussion includes a link to a [StackOverflow question ](https://stackoverflow.com/questions/32028667/are-cmake-source-dir-and-project-source-dir-the-same-in-cmake) that suggests using `add_subdirectory()` to refer to subprojects. JSBSim is now compatible with this CMake feature since the PR #689 has been merged.
